test/hotspot/jtreg/vmTestbase/nsk/share/jvmti/hotswap/HotSwap.cpp

```diff
@@ -67,6 +67,8 @@ static char** names = NULL;
 static jvmtiClassDefinition* old_class_def = NULL;
 static jvmtiClassDefinition* new_class_def = NULL;
 static int classCount = 0;
+/* lock to access classCount */
+static jrawMonitorID classLoadLock = NULL;
```

Comment:
Do you mean use <atomic> from STL?
I'm not sure it would be simpler as we need logic like "get the value, if the value is less than max_classes, increment it"